nvflare/lighter/impl/cert.py

```diff
@@ -22,9 +22,79 @@
 from cryptography.hazmat.primitives.asymmetric import rsa
 from cryptography.x509.oid import NameOID
 
-from nvflare.lighter.spec import Builder, Participant
+from nvflare.lighter.constants import CtxKey
+from nvflare.lighter.spec import Builder, Participant, ParticipantType, Project, PropKey, ProvisionContext
 from nvflare.lighter.utils import serialize_cert, serialize_pri_key
 
+_CERT_BASE_NAME_CLIENT = "client"
+_CERT_BASE_NAME_SERVER = "server"
+_CERT_BASE_NAME_OVERSEER = "overseer"
+
+
+class _CertState:
+
+    CERT_STATE_FILE = "cert.json"
+
+    PROP_ROOT_CERT = CtxKey.ROOT_CERT
+    PROP_ROOT_PRI_KEY = CtxKey.ROOT_PRI_KEY
+    PROP_CERT = "cert"
+    PROP_PRI_KEY = "pri_key"
+
+    def __init__(self, state_dir: str):
+        self.is_available = False
+        self.state_dir = state_dir
+        self.content = {}
+        cert_file = os.path.join(state_dir, self.CERT_STATE_FILE)
+        if os.path.exists(cert_file):
+            self.is_available = True
+            with open(cert_file, "rt") as f:
+                self.content.update(json.load(f))
+
+    def get_root_cert(self):
+        return self.content.get(self.PROP_ROOT_CERT)
+
+    def set_root_cert(self, cert):
+        self.content[self.PROP_ROOT_CERT] = cert
+
+    def get_root_pri_key(self):
+        return self.content.get(self.PROP_ROOT_PRI_KEY)
+
+    def set_root_pri_key(self, key):
+        self.content[self.PROP_ROOT_PRI_KEY] = key
+
+    def has_subject(self, subject: str):
+        return subject in self.content
+
+    def _add_subject_prop(self, subject: str, key: str, value):
+        subject_data = self.content.get(subject)
+        if not subject_data:
+            subject_data = {}
+            self.content[subject] = subject_data
+        subject_data[key] = value
+
+    def _get_subject_prop(self, subject: str, key: str):
+        subject_data = self.content.get(subject)
+        if not subject_data:
+            return None
```

Comment:
no sure if it's worthy to build a default value return here, just like the Python dict get().

Reply:
Could be, but right now there is no use case for that.